### ISSUE TITLE
Cease and Desist

### ISSUE THREAD

**Issue by lmoureaux** [Lexxie9952/fcw.org-server#168]
Dear operators of the website at `https://www.freecivweb.org/`,

It has come to my attention that you make an unauthorized use of my copyrighted work entitled "Network protocol: turn unit orders into array of struct" (the "Work") in the preparation of a work derived therefrom. I have reserved all rights in the Work, which was first published around Sep 3, 2019 in the Freeciv issue tracker at [hostedredmine.com](https://www.hostedredmine.com/issues/833667). It was then incorporated in the Freeciv distribution as commit https://github.com/freeciv/freeciv/commit/e32e134c64dfcec56cd435804eab6c565bf1c648, under the conditions of the "General Public License, either the version 2 of the license, or any later version".

Your work entitled "Bring in unit orders network protocol simplification" (the "Derived Work") and which appears as commit https://github.com/Lexxie9952/fcw.org-server/commit/f95bcb76541f3e5567adff87cd0e5f28550947f9 the code repository at https://github.com/Lexxie9952/fcw.org-server (the "Repository") is essentially identical to the Work and clearly used the Work as its basis. You appear to make use of Section 13 of the General Public License, version 3, to publish the Derived Work under the GNU Affero General Public License (the "GNU AGPL") in the Repository. This establishes that the Work is licensed by me to you under the terms of the GNU AGPL. You neither asked nor received permission to use the Work in any other way.

Section 13 of the GNU AGPL states the following:

> Notwithstanding any other provision of this License, if you modify the Program, your modified version must prominently offer all users interacting with it remotely through a computer network (if your version supports such interaction) an opportunity to receive the Corresponding Source of your version by providing access to the Corresponding Source from a network server at no charge, through some standard or customary means of facilitating copying of software. This Corresponding Source shall include the Corresponding Source for any work covered by version 3 of the GNU General Public License that is incorporated pursuant to the following paragraph.
> [...]

The website at `https://www.freecivweb.org/` (the "Website") is based on code in the Repository, as is acknowledged by the two links called "Development" and "Developers" in its home page. However, the Website contains modifications that are not present in the Repository, in violation of the GNU AGPL. This is further substantiated by the new statement added in commit
https://github.com/Lexxie9952/fcw.org-server/commit/63db41ed5cead14c1efa1bf21ce73970130a41af of the Repository, which states that:

> Regarding potential developers and non-users, FCW is NOT required to do work for, nor release source code to anyone with whom we have not entered a voluntary mutual relationship, nor do we enter into a developmental, provider, or usership relation
with entities who do not agree to the rights we reserve for our absolutely free and unencumbered operation.
Any relation with us conditional to the above.

This is in flagrant contradiction with the requirements of the GNU AGPL under which you use my copyrighted Work. Therefore, I believe you have willfully infringed my rights under 17 U.S.C. Section 101 et seq. and could be liable for statutory damages as high as $150,000 as set forth in Section 504(c) therein.

I demand that you immediately comply with the GNU AGPL or cease the use and distribution of all infringing works derived from 
the Work and that you desist from this or any infringement of my rights in the future. As set forth in Section 8 of the GNU AGPL, your license will be reinstated if you cure the violation prior to 30 days after your receipt of this notice:

> Moreover, your license from a particular copyright holder is reinstated permanently if the copyright holder notifies you of the violation by some reasonable means, this is the first time you have received notice of violation of this License (for any work) from that copyright holder, and you cure the violation prior to 30 days after your receipt of the notice.

If I have not received an affirmative response from you by 01.11.2021 indicating that you have fully complied with these requirements, I shall consider taking any and all legal remedies available to rectify this situation.

Sincerely,
@lmoureaux.

**Comment by Lexxie9952** [comment]
Thank you for your concern.

In part one we will respond directly to certain statements we belief are false, and will need more definitive clarifications before we can consider them as accurate items to be acted upon.

"You appear to make use of Section 13 of the General Public License, version 3, to publish the Derived Work under the GNU Affero General Public License (the "GNU AGPL") in the Repository." 

This is not the case. FCW is a GPL2 LICENSEE of Freeciv server and an AGPL LICENSOR of the web client. Under no circumstance whatsoever are we an AGPL LICENSEE of GPL code. We do have AGPL licensees affiliated to the project. For example, tacticsandtriumph.com is an AGPL licensee which lacks on its repository, changes to its AGPL code for which we are the licensor. If you seek to rectify cases such as this, please start your hunt there.

"This establishes that the Work is licensed by me to you under the terms of the GNU AGPL. You neither asked nor received permission to use the Work in any other way." 

It certainly does not establish that. The work in question, was submitted to the Freeciv project under GPL and was sourced in a way requiring no permission. The commit was made to our repository by a maintainer of the Freeciv project @cazfi, under his full rights of the GPL license. If you believe it is not the case that this code is now copyleft GPL licensed in the Freeciv project, we will react to that information after that be confirmed, but for now there is no evidence to disprove our belief that the code in question is open source GPL licensing, copyleft, and that @cazfi was fully in his rights to submit it to our repository without seeking your permission.

In part two we will summarize our general response to this complaint.

1. You appear to confound the distinction between a GPL licensee and an AGPL licensor, and conflate this into statements representing yourself as an AGPL licensor to FCW who is your AGPL licensee. It seems clear that none of these appear to be the case, whether for the license or the correct licensor/licensee relationship.  
2.  Please delineate the position you are taking for each of the four different licensing statuses with respect to you, your code, the maintainers of the project, and the code at FCW. To clarify, we are asking for (GPL, AGPL) x (licensor, licensee) x (your code in question, FCW code). We believe knowing your stance on these 8 data points leads to better clarification and resolution of any claims of the parties.
3. You appear to have confused your GPL2 code which we license for a completely different project (Freeciv Server), as making you an AGPL licensor of the FCW client to its own creators. An AGPL client merely by connecting to a server, does not make a server AGPL. Unless you would also claim that every AGPL browser causes every server it ever accesses, to instantly become AGPL.
4. The AGPL has no stipulation that licensees cannot privately use it in a private community, and as such only be required to supply source internally to that private community. Indeed it allows that. But even if that is the case, it especially has no stipulation against this for LICENSORS. Indeed, licensors can re-license and re-release the project and even charge a fee to sell the code for profit. Membership in our community entails agreement with our Terms of Service. USERS of our service are indeed entitled to source code free of charge, as per stipulations which you have clearly read since you quote them in your complaint. There is no stipulation by AGPL that a licensor is forced to enter usership relations publicly and at large against its will, and indeed we do not. You are not a member or user of our community, so it is only voluntarily that we allow you to view our public repository free of charge, which you may.
5. There are numerous other inaccuracies in your claims, which I would be happy to have our legal representation discuss with your legal representation, if you provide us with the contact information.
6. Lastly, since you have not contacted or requested any source code from us, you cannot make the claim we don't supply source code. Indeed, the repository on which you make your complaint is the site that has all content required of us under our licensing provisions.  

Undoubtedly, some mistake has occurred. We believe your intention is positive open source collaboration and you respect that we have a normal development cycle, conduct beta testing within our private community, and you respect our free and unencumbered development operations. Assuming this to be the case, I can assure you that there are no licensing issues and invite you to enjoy the positive win/win relations of open source collaboration.

Best Regards,
Lexxie9952

**Comment by lmoureaux** [comment]
Dear @Lexxie9952,

Thank you for your response. Let me clarify the exact lines of code in the present state of the repository (commit 02325fc448e7e83dc200901e43054bd9d769d189) upon which I claim copyright and the basis for this claim. This copyright claim is only intended to be a sufficient basis to prove that I own copyright in various parts of the Repository and is not intended to be complete.

I claim copyright on lines 678 to 684 of the file `freeciv-web/src/main/webapp/javascript/action_dialog.js`:

https://github.com/Lexxie9952/fcw.org-server/blob/02325fc448e7e83dc200901e43054bd9d769d189/freeciv-web/src/main/webapp/javascript/action_dialog.js#L678-L684

These lines are a direct translation from C to JavaScript (by @cazfi) of my Work contributed to https://github.com/freeciv/freeciv/commit/e32e134c64dfcec56cd435804eab6c565bf1c648 under "version 2 of the GNU General Public License, or any later version" ("GPLv2+"). As is usual for translations, the copyright for these lines is shared between the original author (me) and the translator (cazfi). In the Repository, they are indicated as being licensed under the AGPL, which I allowed in Section 13 of version 3 of the GPL.

I similarly claim copyright on the lines in `freeciv-web/src/main/webapp/javascript/control.js` that are direct translations of my Work.

I also claim copyright on the lines in the directory `freeciv/freeciv/` that are part of my Work. I believe that you are also obligated to release your modifications to these because you use them in tight combination with AGPL-licensed code, as mentionned in Section 13 of the AGPL. This is not required, however, to establish my copyright claim on code served at `https://www.freecivweb.org`.

Following the above and to answer your points 1, 2, and 3, I am in the situation of the Licensor, to you, of the Work covered by the GPLv2+ and AGPL licences, and you are in the situation of the Licensee at least for the lines mentioned above. Please notice that the AGPL does not allow sub-licensing and requires the complete Corresponding Source, including build instructions, to be published. I am also in the situation of the Licensee from you for some other parts of the code, but this does not extinguish your obligations under the GPLv2+ and the AGPL for the parts for which you are the Licensee.

Regarding your point 4, it is obvious that the website at `https://www.freecivweb.org` is publicly accessible and is therefore not a "private community", whatever this means. I was able to download a copy of my copyrighted work in binary form from that website.

Please let me know the details of the inaccuracies mentioned in your point 5 so I can examine them.

Regarding your point 5, please note that the AGPL states the following:

> e) Convey the object code using peer-to-peer transmission, provided you inform other peers where the object code and Corresponding Source of the work are being offered to the general public at no charge under subsection 6d.

It is thus clearly required that you inform any other peer, not just me, where the Corresponding Source is being offered. I have been informed that you dismissed a direct request to provide the Corresponding Source. If that were not enough, your statement in commit https://github.com/Lexxie9952/fcw.org-server/commit/63db41ed5cead14c1efa1bf21ce73970130a41af, quoted in my original request, also makes it clear that you refuse to provide the Corresponding Source, in violation of the AGPL.

Please resolve the situation as soon as possible, either by stopping the use of my copyrighted Work, or by complying to the provisions of the AGPL.

Sincerely,
@lmoureaux.

**Comment by Lexxie9952** [comment]
To whom it may concern,

Thank you for the response and attempted clarifications of your position. 

I. In part one we will discuss our concerns over unspoken motives behind this complaint:

We are concerned about your interpretations of licensing which seem verisimilitudinously "engineered" or "constructed" to serve some other motives. We believe it may lead to quicker resolution and satisfaction of all parties to "cut to the chase" and negotiate those motives more openly and directly with you, if you have the good will on your part to do so. We sincerely hope there is no hostile intent and that you have simply made a mistake of judgment, but with the good motive of trying to champion open source collaboration and code sharing. 

II. In part two we will discuss some dubious matters which if resolved, we believe will productively expedite a resolution to this matter. This is because, in our humble opinion, these "dubious matters" are the cause and source of the entire situation. 

1. First, we request that the repository for the site whose operator reports you work on, https://www.tacticsandtriumph.com, be made AGPL compliant, _specifically_, as your licensor we request that it prominently and easily offer all required content for that site, in synch with the actual state of the site at any time a user interacts with it. The operator is currently dodging this by saying he fulfills requests for source code, then linking to incomplete code that does not represent a synchronous snapshot of the state of the site. This is exacerbated by the communications not taking place anywhere on the site itself, so that users who do not know where the operator is making these offers to request the code, have no idea that the content they see on the repository is deliberately NOT the actual content being used by the site. This actually requires more work on your parts, to provide such obstructed means of accessing the code, than by fulfilling the obligation to prominently offer actual and correct data that does not require interaction or the users to somehow know they must make "requests", without any means of knowing so, and no information that the repository they view is not the actual real state of the server. The combination of those four things appears to lend resounding evidence and indefatigable credence to the veracity of another matter that came to our attention: 
2. It came to our attention, a deliberate intent or professed strategy of competition flagrantly against the open source spirit. We would characterize this anti-open strategy in quotations as something like "leverage the 20x more work done at FCW than our own site, while simultaneously adding 1x amount of proprietary and hidden work inaccessible to our licensor, in order to compete against our licensor by presenting a superior product to our licensor while doing only 1/20th the amount of work."  We are certain that anyone pursuing such a strategy in an open source community would be a grave concern to you, and we are certain you will champion open source by giving cease and desist notifications to any party engaging in such practices. Therefore, a large step toward final resolution of this matter will be fulfilled when you rectify that asymmetry. That is, you will make it so that:
A. There is symmetry in the AGPL demands being made for that site, so that it does not give the appearance you are hypocritically leveraging licensing demands on one site but not the other. Specifically, that there is no way an impartial party could view the state of affairs and interpret it like this: "in order to promote the interests of the licensee site over its licensor, access all open source content from the licensor and make licensing demands on the licensor to access this content; while simultaneously NOT fulfilling requirements of the licensee, making no demand that they do so, and putting up excuses to obstruct and block prominent accessibility of the licensee's modifications." Obviously no one would like impartial parties to be able to interpret such appearances. Thus we ask for symmetric licensing demands on all licensees.
B. The site prominently offers its users content representing the actual state of the server
3. As our AGPL licensee, when we no longer see "asymmetric hypocrisy" in the positions you take, we will much more easily be convinced that we are not dealing with a case of what can be legally categorized as "frivolous harassment", and will be much more inclined toward a positive collaboration between our related projects.

III. In part three we will address the copyright and licensing claims you made in your above reply.

1. It appears you are claiming that code you claim as "translated" from a GPL project into an AGPL project makes you an AGPL licensor to that project, rather than a GPL licensor to that project. Please state whatever arguments you have for this, since the license itself clearly does not make this the case. FCW maintains that there is NO code translated from a GPL project into any AGPL project of ours, which is authored by your person; moreover, we claim that you are not in any manner, an AGPL licensor to FCW. Being an AGPL licensor of FCW is NOT required to get our source code, however, so perhaps you are taking an unnecessarily contentious strategy with respect to trying to prove AGPL licensorship.
2. Very specifically, the Freeciv Server project which is separate from the AGPL client, is entirely in optative GPL; your claim that we re-release Freeciv server as AGPL is fictitious.
3. In your "copyright by translation" claim, you are making a well-known statement with historical precedent, that can be quoted as something similar to "any use of a communication protocol is tantamount to translation." Please consult with your legal representation on matters of protocols and code translation. Case precedent is clearly AGAINST creators of a protocol leveraging the protocol itself to make copyright claims against code in projects that must use that protocol, and _this specifically applies to claims where protocol **simplistic** data assignments into data structures obligatorily required within the protocol, are claimed to be "code translations."_ We are gravely concerned you would make such disingenuous claims that have already been determined by precedent and labelled as "bad faith claims." We immediately request that you retract all bad faith claims, for us to move forward toward resolution.
4. Use of a protocol with original instance coding in GPL, in separate instances with separate licensing, does not entitle the original protocol creator to have any copyright claims on code using that protocol.
5. Use of a protocol with original instance coding in GPL, in separate instances with separate licensing, does not entitle the original protocol creator to be a LICENSOR of code who did not write, in the other project, nor inherit priority over the real licensor's license in order to put the licensor into a licensee relation to himself. 
6. A licensee who accepts a dual-option GPL license (e.g., a GPLv2 or GPLv3 license) does not entitle the licensor to SELECT for the licensee whether the licensee opts for GPLv2 or GPLv3.
7. Our attempts to confirm whether you wrote ANY code in control.js, came out negative, nor is there any evidence of translation of any c-client code written by you, in the javascript client.
8. A piece of code doing the same task A as other code doing the same task A, does not in itself qualify as "translation".  This is especially the case for making simple assignments rather than more advanced algorithmic methodologies. Please consult your legal representation for clarification on this matter as we will no longer accept spurious arguments on translation that go against our own advisement on that matter.

Since you have threatened legal action and have begun to incur costs to FCW, you are hereby notified that we will begin meticulously tabulating those costs in time and other costs, such that if this goes to a legal resolution, we will be entitled to seek remedy for those costs, should your claims be found to be frivolous and especially, were not made with legal advisement informing you that your claims were genuinely arguable and not speciously constructed.

With that said, we are hopeful you will address the major points that we believe can proceed to resolution. Which we'll review here:
A. Proceeding to negotiate the real unspoken underlying motives behind these actions, in good faith, toward a win/win resolution for all parties.
B. Cessation of all asymmetry in licensing demands. Universality and consistency of all licensing demands made on all parties. e.g., you do not make demand X of party A who argues that demand is not valid, while at the same time waiving demand X from party B who is flagrantly noncompliant, nor give the appearance of selectively obeying GPL/AGPL licensing only in cases that hinder one party while selectively disobeying GPL/AGPL only in cases that promote one party over the other.

We are optimistic you will be a champion of open source collaboration and sharing and respect that the original cause of this situation is out of our control, but to some degree in your control, such that you have the power to resolve these matters easily should you have a congenial and collaborative desire to do so.

Cheers,
Lexxie9952

**Comment by Canik05** [comment]
https://www.udrop.com/5Zfw/tnt.com-docked-data-10-21.zip

Does in fact reflect the current state of https://www.tacticsandtriumph.com with entirely complete code minus only the config folder, which is the same as yours, only with password information which can not be shared due to security and is not required to be shared. insert your own config folder and you have all the TnT.com code in it's entirety.

Freeciv TnT fully complies with it's AGPL obligations and access to our source code is quiet easy. Despite TnT already meeting with compliance we do plan to go above and beyond making our source even more visible and easily accessible in the near future. 

We have released our current state in full we only request you do the same. Nothing more nothing less.

Freeciv TnT desires amicable and mutually beneficial relations in an open-source community. We are willing to work very closely. We would be open to mutual links between all Freeciv sites. Perhaps even merge sites. However even minimal cooperation would be much appreciated. 

Sincerely,
Canik

**Comment by Canik05** [comment]
Here is the current state of val.tacticsandtriumph.com as well: https://www.udrop.com/5ZkX/tnt.com-valhalla-10-21.zip

Louis has been very kind and helpful towards Tactics & Triumph showing a spirit of cooperation. An open-source spirit. He knows full-well I would give him TnT's source code upon request just as I would anyone else. Including you, Lexxie. anyone. I don't think he can say the same for FCW perhaps this is why his treatment of our sites is different. 

Lexxie, please try to be friends and join with the whole community. We'll all be better for it. There are many talented people with lt.net and tnt.com that I'm sure you would benefit from being friendly with.

**Comment by Lexxie9952** [comment]
It's not how to make friends. You can't make a different zip file every time you do a change and expect users to know about requesting it. This is only a valid state for whatever finite moment in time. Read more carefully our position. Compliance requires prominent accessibility to all users. I'm not going through this again until your workflow is set up that this never happens again.

**Comment by Canik05** [comment]
I'm pretty sure it isn't in the AGPL license that I must have a github repo workflow or that it must be 'prominently accessible'. sharing via uDrop is fine so long as I keep it up to date with new code and keep it publicly available. That being said, I intend to move to github in the future and will increase accessibility asap. uDrop was just a temporary solution. Please be understanding TnT.com has only been operating for about 2 months and I had to learn and do virtually everything myself. I will get to github soon.

You can 'not go through this again' until my workflow is set up but to be clear; 
I do not speak for Louis. He is not a TnT operator. His cease and desist is a separate issue.

**Comment by Lexxie9952** [comment]
You're not compliant. We are. And clearly you are working together. In other private conversations I have told you that I have 0 (zero) time to dedicate to running two servers, this includes any attempts by you to "steal" my time out of our normal workflow processes. Indeed I don't really have time to talk to you except in the case of any short dialogue that leads to a clear win/win result. However, past accords and agreements on this proved excessively wasteful of time with no productive outcome, were disingenuous, and you did everything opposite of what you agreed, on every point important to us, doing the exact outcome we said we would not tolerate. This includes libel and slander and politics against us in communication forums. In this respect you're now excommunicated, a maverick, and 100% on your own with no collaboration from your licensor. Like anyone, you may view our public repository in whatever state is its current workflow, and keep your own operation compliant; and under this more limited model both sites may occasionally gain from each other.

Regards

**Comment by lmoureaux** [comment]
Dear @Lexxie9952,

This message will be short and to-the-point.

First of all, let me point out that it is wrong to assume that 1- the situations of `freecivweb.org` and `tacticsandtriumph.com` are identical and 2- because you don't know about a request it doesn't exist. I believe that this solves all the "asymmetry" claims that you made above, namely part II of your last reply to me. I am unfortunately unable to comment on part I because I don't understand what you are talking about.

In your previous message, you wrote the following:

> Being an AGPL licensor of FCW is NOT required to get our source code, however, so perhaps you are taking an unnecessarily contentious strategy with respect to trying to prove AGPL licensorship.

Thank you for this offer for providing the source code. Sharing the code currently in use at `freecivweb.org` -and in doing so restoring compliance with the AGPL- would indeed settle my complaint immediately. I'm thus looking forward to you providing the Corresponding Sources as mandated by the AGPL.

Sincerely,
@lmoureaux.

**Comment by andreasrosdal** [comment]
Freeciv-web is open source and released with the AGPL license. The website at Freecivweb.org is currently violating the AGPL, by not publishing the source code in accordance with the AGPL license.

**Comment by Lexxie9952** [comment]
This is incorrect and moreover, you have in private supported we are in our
AGPL licensor rights and professed "Excellent" as your response to hoe we
deal with it. While I don't think it's "excellent", please make your
position conforming, thank you.

**Comment by cazfi** [comment]
freecivweb.org is AGPL licensee of the original freeciv-web which it was
originally based on, and all the AGPL contributions done to it since.

**Comment by lmoureaux** [comment]
Dear @Lexxie9952,

It is interesting to see that you are deleting comments from other copyright holders supporting my claim. Fortunately I have records of them.

Since a few days have passed and you were active during that period (deleting @cazfi's and @andreasrosdal's comments on this issue, commenting on the merge request at https://github.com/freeciv/freeciv-web/pull/418, posting on the forum at http://forum.freeciv.org/f/viewtopic.php?f=24&p=104131&sid=2b5628d515382d96bb5c3c2d8c418b1b#p104119, ...), I am going to assume that your offer to provide the source code was a false statement. If uploading the source is more difficult than expected, I give you a chance to request more time by replying here within 48 hours.

Side note: if you wish to keep violating the AGPL, please remember that the your rights under the GPLv3 and AGPL are only reinstated automatically if you restore compliance within 30 days of the first notice. After this period or for the GPLv2, you will need to seek my explicit permission to start distributing the program again. This means that trying to hide will only make things worse for you.

Best regards,
@lmoureaux.
